# Chi-Square Test for Independence
## Background
Mizzare Corporation, a well-known brand in the smart home technology market, sells two major products: Smart Thermostats and Smart Lights.
The company recently conducted a customer feedback survey to assess customer satisfaction across different product types.
They classified customer satisfaction into five categories:

•	Very Satisfied

•	Satisfied

•	Neutral

•	Unsatisfied

•	Very Unsatisfied

The company’s main question is:
"Is there a significant relationship between the type of smart home device purchased and the level of customer satisfaction?"
If yes → The type of device influences customer satisfaction.
If no → The differences in satisfaction are likely due to chance.


# Objective
To determine whether there is a statistically significant association between device type (Smart Thermostat / Smart Light) and customer satisfaction level using the Chi-Square Test for Independence.


# 1. Import Required Libraries

In [1]:

#  1. Import Required Libraries
import numpy as np
import pandas as pd
from scipy.stats import chi2_contingency, chi2


# 2. Create the Contingency Table

In [2]:

#  2. Create the Contingency Table
data = np.array([
    [50, 70],   # Very Satisfied
    [80, 100],  # Satisfied
    [60, 90],   # Neutral
    [30, 50],   # Unsatisfied
    [20, 50]    # Very Unsatisfied
])

df = pd.DataFrame(
    data,
    columns=["Smart Thermostat", "Smart Light"],
    index=["Very Satisfied", "Satisfied", "Neutral", "Unsatisfied", "Very Unsatisfied"]
)

print("Contingency Table:")
print(df)


Contingency Table:
                  Smart Thermostat  Smart Light
Very Satisfied                  50           70
Satisfied                       80          100
Neutral                         60           90
Unsatisfied                     30           50
Very Unsatisfied                20           50


# 3. Perform the Chi-Square Test

In [3]:

# 3. Perform Chi-Square Test
chi2_stat, p_val, dof, expected = chi2_contingency(df)

print("\nChi-Square Statistic:", round(chi2_stat, 4))
print("Degrees of Freedom:", dof)
print("P-Value:", round(p_val, 4))



Chi-Square Statistic: 5.6382
Degrees of Freedom: 4
P-Value: 0.2278


## Explanation:
•	Chi-Square Statistic → Measures how different the observed data is from the expected data.

•	Degrees of Freedom (dof) = (rows-1) × (columns-1) = (5-1) × (2-1) = 4.

•	P-value helps determine whether the difference is statistically significant.


# 4. Determine the Critical Value

In [4]:

# 4. Determine Critical Value
alpha = 0.05
critical_value = chi2.ppf(1 - alpha, dof)
print("Critical Value at 0.05 significance:", round(critical_value, 4))


Critical Value at 0.05 significance: 9.4877


# 5. Hypotheses & Final Decision

In [5]:

# 5. Decision with Hypotheses in Final Output

# State Hypotheses
H0 = "H₀: Device type and customer satisfaction are independent."
H1 = "H₁: Device type and customer satisfaction are associated."

print("\n--- Hypotheses ---")
print(H0)
print(H1)

# Decision
if chi2_stat > critical_value:
    decision = "Reject H₀ → Significant association found."
else:
    decision = "Fail to Reject H₀ → No significant association."

print("\n--- Final Decision ---")
print(f"Chi-Square Statistic: {chi2_stat:.4f}")
print(f"Critical Value: {critical_value:.4f}")
print(decision)



--- Hypotheses ---
H₀: Device type and customer satisfaction are independent.
H₁: Device type and customer satisfaction are associated.

--- Final Decision ---
Chi-Square Statistic: 5.6382
Critical Value: 9.4877
Fail to Reject H₀ → No significant association.


# 6. Expected Frequencies

In [6]:

# 6. Expected Frequencies
print("\nExpected Frequencies Table:")
print(pd.DataFrame(expected, index=df.index, columns=df.columns))



Expected Frequencies Table:
                  Smart Thermostat  Smart Light
Very Satisfied                48.0         72.0
Satisfied                     72.0        108.0
Neutral                       60.0         90.0
Unsatisfied                   32.0         48.0
Very Unsatisfied              28.0         42.0


## Explanation:
Expected frequencies represent the counts we would expect if there was no relationship between device type and satisfaction.


# Interpretation
•	Chi-Square Statistic (5.6382) is less than Critical Value (9.4877).

•	P-value (0.2278) is greater than 0.05 significance level.

•	Both tests indicate we Fail to Reject the Null Hypothesis.


# Final Conclusion
Based on the Chi-Square Test for Independence:
•	There is no statistically significant association between device type and customer satisfaction level.
•	Any observed differences in satisfaction could be due to random chance, not because of the product type.
